pynicotine/networkfilter.py

```diff
@@ -407,9 +412,16 @@ def get_online_username(ip_address):
 
     def get_country_code(self, ip_address):
 
-        country_code = self._ip2location.get_country_code(ip_address)
+        try:
+            ip_num, = UINT32_UNPACK(inet_aton(ip_address))
+            ip_index = bisect_left(ip_ranges.values, ip_num)
+            country_code = country_codes.values[ip_index]
 
-        if country_code is None or country_code == "-":
+        except Exception as error:
```

Comment:
Exactly which type of Exception(s) might be expected here?

I only predict ValueError or IndexError, but I don't know about what types of errors the socket or Struct modules can throw.

Reply:
My main concern was invalid IP addresses from the network, but I realized that the message parser would already catch those.